src/input/datetime.rs

```diff
@@ -220,10 +220,12 @@ impl<'a> EitherDateTime<'a> {
     pub fn try_into_py(self, py: Python<'a>) -> PyResult<PyObject> {
         let dt = match self {
             Self::Raw(datetime) => {
-                let tz: Option<PyObject> = match datetime.offset {
+                let tz: Option<&PyTzInfo> = match datetime.offset {
                     Some(offset) => {
                         let tz_info = TzInfo::new(offset);
-                        Some(Py::new(py, tz_info)?.to_object(py))
+                        let obj = Py::new(py, tz_info)?.to_object(py);
```

Comment:
I think the following might work for these three lines:

```suggestion
                        Some(Py::new(py, tz_info)?.to_object(py).extract(py)?)
```

Will rely on type inference for `.extract()` to give you the `&PyTzInfo`.

I do think there's some missing API and documentation for PyO3 in this area, just haven't quite decided what the API should look like.

Reply:
Thank you! I first tried that but had issue with lifetime of variable since it `Py::new` takes ownership of the pointer (hence my usage of `from_borrowed_ptr` instead. I fixed it in [8f09d8c](https://github.com/pydantic/pydantic-core/pull/236/commits/8f09d8ca1626f8b790f597e79336fbad676564d8)